src/query/xasl_cache.c

```diff
@@ -276,6 +291,8 @@ xcache_initialize (THREAD_ENTRY * thread_p)
   xcache_Cleanup_flag = 0;
   xcache_Cleanup_bh = bh_create (thread_p, XCACHE_CLEANUP_NUM_ENTRIES (xcache_Soft_capacity),
 				 sizeof (XCACHE_CLEANUP_CANDIDATE), xcache_compare_cleanup_candidates, NULL);
+  xcache_cleanup_array = malloc (xcache_Soft_capacity * sizeof (XCACHE_CLEANUP_CANDIDATE));
```

Comment:
Please move it to next to `  if (xcache_Cleanup_bh == NULL)`.
The block to allocate `xcache_Cleanup_bh` is very special to use the global private heap.

Please also add error handling of `malloc`.